### PR TITLE
feat(moduleobject): use struct to describe init

### PR DESCRIPTION
**Pull request by Decodetalkers**
in cpp20, struct can use as AStruct { .name = "ss" }, although this project use 17, but a struct can storage more message, this I think will make it easy to construct a moduleobject

Log: use struct to describe init

### CODE REVIEW

**Comment by Groveer** [comment]
Don't use new feature, special cpp20

**Comment by justforlxz** [comment]
大概看了一下，貌似用QVariant也能做到optional吧🙈

**Comment by Decodetalkers** [comment]
> Don't use new feature, special cpp20
只是给用20开发插件的人一个选项而已，我只是觉得用结构体把信息包起来会比较好

File: include/interface/moduleobject.h [hunk 1]
```diff
@@ -45,6 +47,20 @@ enum : DCC_MODULE_TYPE {
...
+/**
+ * @brief ModuleObjectMessage 作为初始化Message的结构体，其中name和displayName 必须有数值
+ * 这个结构体保存了所有初始化的信息
+ */
+
+struct ModuleObjectMessage
+{
+    std::optional<QString> name = std::nullopt;
+    std::optional<QString> displayName = std::nullopt;
+    std::optional<QStringList> contentText = std::nullopt;
```
Comment (18202781743) [inline include/interface/moduleobject.h:59]: 觉得接口尽量不用新特性，这里只用了这个语法而已，感觉没必要

File: src/interface/moduleobject.cpp [hunk 2]
```diff
@@ -93,6 +94,21 @@ ModuleObject::ModuleObject(const QString &name, const QString &displayName, cons
 {
 }
 
+ModuleObject::ModuleObject(const ModuleObjectMessage &message, QObject *parent)
+    : QObject(parent)
```
Comment (18202781743) [inline src/interface/moduleobject.cpp:98]: 这里可以直接调用那个构造函数，对于里面的断言，可以放到那个构造函数里，

File: include/interface/moduleobject.h (hunk 1 above)
Comment (ComixHe) [reply include/interface/moduleobject.h:59]: 感觉直接提供默认值就行，例如这样：
```cpp
struct A{
    QStirng name{"default"};
};
```

**Comment by Decodetalkers** [comment]
> Don't use new feature, special cpp20

好吧，我想了下，确实只有判空的需求，没必要引入

File: include/interface/moduleobject.h [hunk 3]
```diff
@@ -10,6 +10,8 @@
 #include <QObject>
 #include <QVariant>
 
+#include <optional>
```
Comment (18202781743) [inline include/interface/moduleobject.h:13]: ？

**Review by 18202781743 — DISMISSED** [review]

**Review by ComixHe — APPROVED** [review]

**Review by asterwyx — APPROVED** [review]

File: include/interface/moduleobject.h [hunk 4]
```diff
@@ -45,6 +45,20 @@ enum : DCC_MODULE_TYPE {
 
 class ModuleObjectPrivate;
 
+/**
+ * @brief ModuleObjectMessage 作为初始化Message的结构体，其中name和displayName 必须有数值
+ * 这个结构体保存了所有初始化的信息
+ */
+
+struct ModuleObjectMessage
```
Comment (justforlxz) [inline include/interface/moduleobject.h:53]: ModuleInitContext

**Review by justforlxz — CHANGES_REQUESTED** [review]

File: src/interface/moduleobject.cpp [hunk 5]
```diff
@@ -4,6 +4,7 @@
 #include "interface/moduleobject.h"
 
 #include <QWidget>
+#include <QDebug>
```
Comment (justforlxz) [inline src/interface/moduleobject.cpp:7]: 删除

**Review by justforlxz — CHANGES_REQUESTED** [review]

**Review by justforlxz — APPROVED** [review]

**Pull request merged**